packages/SwingSet/src/vats/comms/delivery.js

```diff
@@ -132,14 +132,27 @@ export function makeDeliveryKit(state, syscall, transmit, clistKit, stateKit) {
       // changed to assert here that the result parameter is null or undefined.
       syscall.resolve([[result, false, UNDEFINED]]);
     }
-    const command = message.split(':', 1)[0];
+    // The message is preceded by an optional sequence number:
+    // `$seqnum:$actualMessage` or `:actualMessage`
+    const colon = message.indexOf(':');
+    assert(colon >= 0, X`received message ${message} lacks seqNum delimiter`);
+    const seqNum = message.substring(0, colon);
+    const remote = getRemote(state, remoteID);
+    assert(
+      seqNum === '' || seqNum === `${remote.nextExpectedRecvSeqNum}`,
+      X`unexpected recv seqNum ${seqNum}`,
+    );
+    remote.nextExpectedRecvSeqNum += 1;
+
+    const msgBody = message.substring(colon + 1);
+    const command = msgBody.split(':', 1)[0];
     if (command === 'deliver') {
-      return sendFromRemote(remoteID, message);
+      return sendFromRemote(remoteID, msgBody);
```

Comment:
Not for this PR, but I expect for the next phase we'll be passing `seqNum`(as a BigInt) into `sendFromRemote` and `resolveFromRemote` as a third argument, to support the table comparisons, rather than having them attempt to reference `remote.nextExpectedRecvSeqNum` and deduct one or something.

Reply:
Yes.  I expect that will be needed, in particular for send, since tracking mentions is likely to be intertwined with ID translation.